opm/simulators/linalg/bda/BILU0.hpp

```diff
@@ -0,0 +1,118 @@
+/*
+  Copyright 2019 Equinor ASA
+
+  This file is part of the Open Porous Media project (OPM).
+
+  OPM is free software: you can redistribute it and/or modify
+  it under the terms of the GNU General Public License as published by
+  the Free Software Foundation, either version 3 of the License, or
+  (at your option) any later version.
+
+  OPM is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  GNU General Public License for more details.
+
+  You should have received a copy of the GNU General Public License
+  along with OPM.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+#ifndef BILU0_HPP
+#define BILU0_HPP
+
+#include <config.h> // CMake
+
+#include <memory>
+
+#include <opm/simulators/linalg/bda/BlockedMatrix.hpp>
+
+#if HAVE_OPENCL
+#define __CL_ENABLE_EXCEPTIONS
+#include <CL/cl.hpp>            // up to OpenCL 1.2
```

Comment:
On my system I need to install package opencl-clhpp-headers to get the cl.hpp header. Please test for this file with Cmake and issue a meaningful error message.